resources/snakefiles/mapping.smk

```diff
@@ -0,0 +1,99 @@
+from os.path import splitext
+
+rule index_contigs:
+    input:
+        contigs = lambda wildcards: get_contigs(wildcards.to_sample,
+                                                binning_df)
+    output:
+        temp(multiext("output/binning/indexed/{to_sample}",
+                      ".1.bt2",
+                      ".2.bt2",
+                      ".3.bt2",
+                      ".4.bt2",
+                      ".rev.1.bt2",
+                      ".rev.2.bt2"))
+    log:
+        "output/logs/binning/bowtie2-build.{to_sample}.log"
+    conda:
+        "../env/bowtie2.yaml"
+    params:
+        bt2b_command = config['params']['bowtie2']['bt2b_command'],
+        extra = config['params']['bowtie2']['extra'],  # optional parameters
+        indexbase = "output/binning/indexed/{to_sample}"
+    threads:
+        config['threads']['bowtie2_build']
+    shell:
+        """
+        {params.bt2b_command} --threads {threads} {params.extra} \
+        {input.contigs} {params.indexbase} 2> {log} 1>&2
+        """
+
+rule map_reads:
+    """
+    Performs host read filtering on paired end data using Bowtie and Samtools/
+    BEDtools.
+
+    Also requires an indexed reference (path specified in config).
+
+    First, uses Bowtie output piped through Samtools to only retain read pairs
+    that are never mapped (either concordantly or just singly) to the indexed
+    reference genome. Fastqs from this are gzipped into matched forward and
+    reverse pairs.
+
+    Unpaired forward and reverse reads are simply run through Bowtie and
+    non-mapping gzipped reads output.
+
+    All piped output first written to localscratch to avoid tying up filesystem.
+    """
+    input:
+        reads = lambda wildcards: expand(rules.merge_units.output,
```

Comment:
This looks like the reads that are being mapped are the output of the `merge_units` rule. It seems like the output of the `host_filter` rule would make it more consistent, since those are the reads that the assemblies were generated from.